jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/PathMappingsHandlerTest.java

```diff
@@ -0,0 +1,242 @@
+             
+            """);
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+        assertEquals("PhpExample Hit", response.getContent());
+    }
+
+    /**
+     * Test where there is only a single mapping, and a wrapper for fallback.
+     */
+    @Test
+    public void testOneMappingAndWrapper() throws Exception
+    {
+        ContextHandler contextHandler = new ContextHandler();
+        contextHandler.setContextPath("/");
+
+        PathMappingsHandler pathMappingsHandler = new PathMappingsHandler();
+        pathMappingsHandler.setHandler(new SimpleHandler("WrapperFoo Hit"));
+        pathMappingsHandler.addMapping(new ServletPathSpec("*.php"), new SimpleHandler("PhpExample Hit"));
+        contextHandler.setHandler(pathMappingsHandler);
+
+        startServer(contextHandler);
+
+        HttpTester.Response response = executeRequest("""
+            GET /hello HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+             
+            """);
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+        assertEquals("WrapperFoo Hit", response.getContent());
+
+        response = executeRequest("""
+            GET /hello.php HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+             
+            """);
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+        assertEquals("PhpExample Hit", response.getContent());
+    }
+
+    public static Stream<Arguments> severalMappingsInput()
+    {
+        return Stream.of(
+            Arguments.of("/hello", HttpStatus.OK_200, "FakeResourceHandler Hit"),
+            Arguments.of("/index.html", HttpStatus.OK_200, "FakeSpecificStaticHandler Hit"),
+            Arguments.of("/index.php", HttpStatus.OK_200, "PhpHandler Hit"),
+            Arguments.of("/config.php", HttpStatus.OK_200, "PhpHandler Hit"),
+            Arguments.of("/css/main.css", HttpStatus.OK_200, "FakeResourceHandler Hit")
+        );
+    }
+
+    /**
+     * Test where there are a few mappings, with a root mapping, and no wrapper.
+     * This means the wrapper would not ever be hit, as all inputs would match at
+     * least 1 mapping.
+     */
+    @ParameterizedTest
+    @MethodSource("severalMappingsInput")
+    public void testSeveralMappingAndNoWrapper(String requestPath, int expectedStatus, String expectedResponseBody) throws Exception
+    {
+        ContextHandler contextHandler = new ContextHandler();
+        contextHandler.setContextPath("/");
+
+        PathMappingsHandler pathMappingsHandler = new PathMappingsHandler();
+        pathMappingsHandler.addMapping(new ServletPathSpec("/"), new SimpleHandler("FakeResourceHandler Hit"));
+        pathMappingsHandler.addMapping(new ServletPathSpec("/index.html"), new SimpleHandler("FakeSpecificStaticHandler Hit"));
+        pathMappingsHandler.addMapping(new ServletPathSpec("*.php"), new SimpleHandler("PhpHandler Hit"));
+        contextHandler.setHandler(pathMappingsHandler);
```

Comment:
Example 2 (of 2) that might be useful for FCGI (to PHP) Proxy.
This one relies on default servlet pattern matching behavior to use the resource handler for non-php requests.